apps/bestofjs-nextjs/src/app/hall-of-fame/hall-of-fame-view.client.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import { useState } from "react";
+import { InfoIcon } from "lucide-react";
+
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+
+import { searchHallOfFameMembers } from "./actions";
+import { HallOfFameSearchBar } from "./hall-of-fame-search.client";
+import { HallOfFameMemberList } from "./hall-of-member-list";
+
+export function HallOfFameClientView({
+  initialContent,
+}: {
+  initialContent: React.ReactNode;
+}) {
+  const [query, setQuery] = useState<string>("");
+  const [results, setResults] = useState<BestOfJS.HallOfFameMember[]>([]);
+  const [isPending, setIsPending] = useState(false);
+
+  const onSearch = async (searchQuery: string) => {
+    setQuery(searchQuery);
+    setIsPending(true);
+    const members = await searchHallOfFameMembers(searchQuery);
+    setIsPending(false);
+    setResults(members);
+  };
+
+  const onReset = () => {
+    setQuery("");
+  };
+
+  if (isPending) return <>Searching...</>;
```

Comment:
It would improve the UX to put this `Searching...` UI in the search results section below and disable the searchbar.

Currently it causes the searchbar to disappear and reappear
![image](https://github.com/bestofjs/bestofjs/assets/2955134/9b7bf111-1fc6-41be-b7b2-47d63b940314)

![image](https://github.com/bestofjs/bestofjs/assets/2955134/57675277-c73a-4c6b-be19-66512ec25a50)

Reply:
Good catch, I have improved the UI, to avoid hiding the search input while searching.